align_system/algorithms/lib/algorithmic_decision_maker.py

```diff
@@ -0,0 +1,35 @@
+from abc import abstractmethod
+
+# ADM sub-classes implement all the algorithm-specific logic
+class AlgorithmicDecisionMaker:
+    
+    @abstractmethod
+    def __call__(self, sample, target_kdma_values, **kwargs):
```

Comment:
Need to differentiate between aligned ADM an unaligned (unaligned won't have target_kdma_values; maybe just handling `None`)